### DISCUSSION TITLE
Problems reading a table in the grid

### DISCUSSION THREAD

**Discussion by patriciomh1965** [serenity-is/Serenity#5354] [Help]
![grilla](https://user-images.githubusercontent.com/42218266/104623619-4e4e9c80-5671-11eb-8d0f-484c7ba9f876.png)
Hello community, I have tried to read an information from the table that contains 2 compound keys. But I can't get the information, here is my schedule. If someone knows how to do it, I would appreciate:

**GRID :**

namespace SegurosMAPC.BaseDatos {

    import fld = EleDiarioRow.Fields;

    @Serenity.Decorators.registerClass()
    export class EleDiarioGrid extends Serenity.EntityGrid<EleDiarioRow, any> {
        protected getColumnsKey() { return 'BaseDatos.EleDiario'; }
        protected getDialogType() { return EleDiarioDialog; }
        protected getIdProperty() { return EleDiarioRow.idProperty; }
        protected getInsertPermission() { return EleDiarioRow.insertPermission; }
        protected getLocalTextPrefix() { return EleDiarioRow.localTextPrefix; }
        protected getService() { return EleDiarioService.baseUrl; }

        private pendingChanges: Q.Dictionary<any> = {};

        constructor(container: JQuery) {
            super(container);
        }
.......
........     
        protected getColumns() {
            var columns = super.getColumns();
            columns.splice(1, 0,
                {
                    field: 'EDITAR',
                    name: '',
                    format: ctx => '<a class="inline-action editar" title="Editar la Linea">' +
                        '<i class="fa fa-pencil text-blue"></i></a>',
                    width: 24,
                    minWidth: 24,
                    maxWidth: 24
                });      
            return columns;
        }

        protected onClick(e: JQueryEventObject, row: number, cell: number) {
            super.onClick(e, row, cell);

            if (e.isDefaultPrevented())
                return;

            var item = this.itemAt(row);
            var estadolinea = item.EstadoLinea;
            var estadocargaele = item.EstadoCargaEle;
            var estadomapeo = item.EstadoMapeo;
            var estadoinyeccion = item.EstadoInyeccion;

            if (estadolinea == 6) { Q.notifyError("La linea ya se encuentra con Error, no puede hacer ninguna operación"); return; }  // linea con error

            var target = $(e.target);

            if (target.parent().hasClass('inline-action'))
                target = target.parent();

            if (target.hasClass('inline-action')) {
                e.preventDefault();
                

                // 
                else if (target.hasClass('inline-action')) {
                    e.preventDefault();
                   
                // EDITAR
                if (target.hasClass('editar')) {
                    if (estadocargaele == 2) { Q.notifyWarning("La linea ya se encuentra Aprobada"); return; }  // linea aprobada
                    var evento = item.CodEvento;
                    var materia = item.CodRamo;
                    var aprob = item.EstadoLinea;
                    var valor = item.KeyProLine;

                    if (aprob == 2 || aprob == 3 || aprob == 4 || aprob == 5) {
                        Q.warning("No puede hacer cambios porque no esta en condicion de Pendiente, para hacerlo debe cambiar el estado !!");
                        return;
                    } 

                    //  HERE IS MY PROBLEM
                    Q.reloadLookup(TablaSeguroContratoRow.lookupKey); 
                    var s1 = Q.tryFirst(TablaSeguroContratoRow.getLookup().items, e => e.SegProcNumero == item.ProcesoNumero && e.SegNumLinea == item.NumLinea)
                    if (s1 != null) {
                        Q.warning('dato ' + s1.SegPrimaBruta);
                    }
                    //                        

**ROW :**

namespace SegurosMAPC.BaseDatos.Entities
{
    using Serenity;
    using Serenity.ComponentModel;
    using Serenity.Data;
    using Serenity.Data.Mapping;
    using System;
    using System.Collections.Generic;
    using System.ComponentModel;
    using System.IO;

    [ConnectionKey("BaseDatos"), Module("BaseDatos"), TableName("[dbo].[ta_evento_seguro_contrato]")]
    [DisplayName("Seguros"), InstanceName("Seguros")]
    [ReadPermission("Administration:General")]
    [ModifyPermission("Administration:General")]
    
    [LookupScript(Expiration = -1)]    
    public sealed class TablaSeguroContratoRow : Row, IIdRow, INameRow
    {

        [DisplayName("Key Proline"), Column("key_proline"), Identity]
        public Int32? KeyProline
        {
            get { return Fields.KeyProline[this]; }
            set { Fields.KeyProline[this] = value; }
        }

        [DisplayName("Numero Proceso"), Column("seg_proc_numero"), PrimaryKey]
        public Int32? SegProcNumero
        {
            get { return Fields.SegProcNumero[this]; }
            set { Fields.SegProcNumero[this] = value; }
        }

        [DisplayName("Numero Linea"), Column("seg_num_linea"), PrimaryKey]
        public Int32? SegNumLinea
        {
            get { return Fields.SegNumLinea[this]; }
            set { Fields.SegNumLinea[this] = value; }
        }

        [DisplayName("Prima Bruta"), Column("seg_prima_bruta"), LookupInclude]
        public Int32? SegPrimaBruta
        {
            get { return Fields.SegPrimaBruta[this]; }
            set { Fields.SegPrimaBruta[this] = value; }
        }
.....
.....

**DATA BASE TABLE:**                  
![table](https://user-images.githubusercontent.com/42218266/104625695-c7e78a00-5673-11eb-88e8-8ed907d38d07.png)

Variable s1 is null, however the record exists in table.

![registro](https://user-images.githubusercontent.com/42218266/104627686-1dbd3180-5676-11eb-97c1-ca1a1cd536e3.png)

**Comment by edwardch** [comment]
I see in TablaSeguroContratoRow.cs you have `[DisplayName("Key Proline"), Column("key_proline"), Identity]`, meaning that Serenity will treat key_proline as the ID column. 

Can you go to your column picker and restore defaults maybe?
